Source/Scene/Batched3DModel3DTileContent.js

```diff
@@ -302,6 +341,59 @@ define([
         return false;
     };
 
+    function cloneShaderProgram(frameState, shaderProgram, modifyFragmentShaderCallback) {
+        var context = frameState.context;
+        var attributeLocations = shaderProgram._attributeLocations;
+        var vs = shaderProgram.vertexShaderSource.sources[0];
+        var fs = shaderProgram.fragmentShaderSource.sources[0];
+        if (defined(modifyFragmentShaderCallback)) {
+            fs = modifyFragmentShaderCallback(fs);
+        }
+        return ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : vs,
+            fragmentShaderSource : fs,
+            attributeLocations : attributeLocations
+        });
+    }
+
+    function updateColorBlendMode(content, frameState, colorBlendMode) {
```

Comment:
This workflow is pretty complicated.  Perhaps there's some renderer changes that could make it easier.  Let's discuss offline.